sme_management/views.py

```diff
@@ -23,3 +32,121 @@ class ManageSMEUserView(generics.RetrieveUpdateAPIView):
     def get_queryset(self):
         """Retrieve the SMEUser for the authenticated user"""
         return self.queryset.filter(user=self.request.user)
+
+
+class SMEProjectAPIView(APIView):
+    """
+    API View to create projects with documents -
+    cashflow statement, income statement &
+    balance sheets
+    """
+
+    authentication_classes = [TokenAuthentication, ]
+    permission_classes = [IsAuthenticated, ]
+    serializer_class = SMEProjectSerializer
+    parser_class = (FileUploadParser,)
+    search_fields = ('status',)
+
+    def post(self, request, *args, **kwargs):
+
+        incomplete_project_exists = SMEProject.objects.filter(~Q(status='COMPLETED')).exists()
+        if incomplete_project_exists:
+            return Response(data="An incomplete project still exists",
+                            status=status.HTTP_400_BAD_REQUEST)
+
+        serializer = self.serializer_class(data=request.data)
+
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+        else:
+            print(serializer.errors)
+            return Response(serializer.data, status=status.HTTP_400_BAD_REQUEST)
+
+    # TODO: In the future, we would want to make a separation between
+    #  how we get COMPLETED, IN_PROGRESS and UNAPPROVED projects
+    #  For now, we just return everything in one call
+    #  There might be a solution to this with search_fields. Confirm
+    def get(self, request, format=None):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Redefining built-in 'format'](https://app.codacy.com/gh/BuildForSDG/Team-004-Backend/pullRequest?prid=5635476)